src/ext/java/org/opentripplanner/ext/siri/AddedTripBuilder.java

```diff
@@ -216,6 +216,11 @@ Result<TripUpdate, UpdateError> build() {
       aimedStopTimes,
       transitModel.getDeduplicator()
     );
+    // validate the scheduled trip times
+    // they are in general superseded by real-time trip times
+    // but in case of trip cancellation, OTP will fall back to scheduled trip times
+    // therefore they must be valid
+    tripTimes.validateNonIncreasingTimes();
```

Comment:
We should discuss if we should even allow creating invalid trip times or if this check should happen in the constructor.

Reply:
Absolutely. There are however tricky corner cases with cancelled stops:  these stops are marked as "cancelled" (PickDrop.CANCELLED), their estimated times are no longer updated and ignored in the routing logic. They may become invalid (travel back in time), so a systematic validation of TripTimes should take this into account.

Reply:
Be biggest problem here is the FLEX trips - witch should not use this, but their own implementation. But, until they do, we need to call this explicit. There might also be a separate RT version - at least an extension.

Reply:
Note! We should solve this by tagging each tripTime with the "status". CANCELED_TRIP, CANCELED_STOP, CANCELED_STOP+OTP_INTERPOLATED.